ride_share.rb

```diff
@@ -0,0 +1,135 @@
+#Structure of my_company datas looks like this:
+my_company = [
+  {
+    DR0001: 
+    [
+      {
+        date: "3rd Feb 2016",
+        cost: 10,
+        ride_id: "RD0003",
+        rating: 3
+      },
+      {
+        date: "3rd Feb 2016",
+        cost: 30,
+        ride_id:"RD0015",
+        rating: 4
+      },
+      {
+        date: "6th Feb 2016",
+        cost: 45,
+        ride_id: "RD0003",
+        rating: 2
+      }
+    ],
+  },
+ {
+   DR0002: 
+   [
+     {
+       date: "3rd Feb 2016",
+       cost: 25,
+       ride_id: "RD0073",
+       rating:5
+      },
+      {
+        date: "4th Feb 2016",
+        cost: 15,
+        ride_id: "RD0013",
+        rating:1
+      },
+      {
+        date: "7th Feb 2016",
+        cost: 35,
+        ride_id: "RD0066",
+        rating:3
+        }
+    ],
+  },
+ {
+   DR0003: 
+   [
+     {
+       date: "4th Feb 2016",
+       cost: 5,
+       ride_id: "RD0066",
+       rating:5
+      },
+      {
+        date: "5th Feb 2016",
+        cost: 50,
+        ride_id: "RD0003",
+        rating:2}
+    ],
+  },
+ {DR0004:
+  [
+    {
+      date: "3rd Feb 2016",
+      cost: 5,
+      ride_id: "RD0022",
+      rating:5
+    },
+    {
+      date: "4th Feb 2016",
+      cost: 10,
+      ride_id: "RD0022",
+      rating:4
+    },
+      {date: "8th Feb 2016",
+       cost: 20,
+       ride_id: "RD0073",
+       rating:5
+      }
+    ]
+  }
+]
+
+# Total of RIDES, EARNINGS , AVERAGE RATING, MOST EARNING PER DATE for each driver.
+# Max of earnin and averge rating from all the drivers
+def statistic_for_each_driver(my_company)
+  drivers_list = []
+  drivers_total_earnings = []
+  drivers_avg_ratings = []
+  my_company.each do |drivers|
```

Comment:
It's good that you're using a method, but it might be more efficient and meaningful to use methods to break down this problem into smaller chunks, rather that having a single big method that does all of the work you would otherwise do.